free/src/main/scala/cats/free/Free.scala

```diff
@@ -266,6 +267,25 @@ object Free extends FreeInstances {
       Free.liftF(I.inj(fa))
   }
 
+  /**
+    * This method is used to defer the application of an InjectK[F, G]
+    * instance. The actual work happens in
+    * `FreeLiftInjectKPartiallyApplied#apply`.
+    *
+    * This method exists to allow the `G` parameter to be
+    * bound independently of the `F` and `A` parameters below.
+    */
+  def liftInject[G[_]]: FreeLiftInjectKPartiallyApplied[G] =
+    new FreeLiftInjectKPartiallyApplied
+
+  /**
+    * Uses the [[http://typelevel.org/cats/guidelines.html#partially-applied-type-params Partially Applied Type Params technique]] for ergonomics.
+    */
+  private[free] final class FreeLiftInjectKPartiallyApplied[G[_]](val dummy: Boolean = true ) extends AnyVal {
```

Comment:
I'm surprised that it's not a problem to use this as a public return type when the class is package private. It looks like this is the pattern that we are using elsewhere, though. I guess since it's a value class that gets erased it ends up not mattering?